File: gherkin/dotnet/Gherkin/CucumberMessages/Types/Scenario.cs
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace Gherkin.CucumberMessages.Types
{
    public class Scenario
    {
        [DataMember(Name = "location")]
        public Location Location { get; set; }

        [DataMember(Name = "tags")]
        public IReadOnlyCollection<Tag> Tags { get; set; }

        [DataMember(Name = "keyword")]
        public string Keyword { get; set; }

        [DataMember(Name = "name")]
        public string Name { get; set; }

        [DataMember(Name = "description")]
        public string Description { get; set; }

        [DataMember(Name = "steps")]
        public IReadOnlyCollection<Step> Steps { get; set; }

        [DataMember(Name = "examples")]
        public IReadOnlyCollection<Examples> Examples { get; set; }

        [DataMember(Name = "id")]
        public string Id { get; set; }
    }
}

File: gherkin/dart/lib/src/ast/Scenario.dart
import 'package:gherkin/core.dart';
import 'package:gherkin/extensions.dart';
import 'package:gherkin/language.dart';

import 'Examples.dart';
import 'IHasTags.dart';
import 'Step.dart';
import 'StepsContainer.dart';
import 'Tag.dart';

class Scenario extends StepsContainer implements IHasTags, INullSafetyObject
{
  static const empty = _InvalidScenario();

  final Iterable<Tag> _tags;

  final String id;

  @override
  Iterable<Tag> get tags => _tags;

  final Iterable<Examples> examples;

  const Scenario(this._tags, Location location, String keyword, String name
      , String description, List<Step> steps, this.examples
      , {this.id=Strings.empty})
    : super(location, keyword, name, description, steps);

  @override
  bool get isEmpty => false;
}

/// Convenience implementation of an invalid [Scenario] instance.
class _InvalidScenario extends Scenario
{
  const _InvalidScenario()
    : super(const <Tag>[], Location.empty, Strings.empty
      , Strings.empty, Strings.empty, const <Step>[], const <Examples>[]);

  @override
  bool get isEmpty => true;
}
